File: bun/src/repositories/category-repository.ts
import type { Category } from '../models/category.js';

export class CategoryRepository {
  private categories: Category[] = [
    { id: '1', name: 'Dairy' },
    { id: '2', name: 'Fruit' },
    { id: '3', name: 'Vegetables' },
    { id: '4', name: 'Bakery' },
    { id: '5', name: 'Meat' },
  ];

  getAllCategories(): Category[] {
    return this.categories;
  }

  getCategoryById(id: string): Category {
    return this.categories.find(c => c.id === id)!;
  }
}

File: aspnetcore/Repositories/CategoryRepository.cs
using local_ly_dotnet.Models;

namespace local_ly_dotnet.Repositories;

public class CategoryRepository
{
    private readonly List<Category> _categories =
    [
        new Category { Id = "1", Name = "Dairy" },
        new Category { Id = "2", Name = "Fruit" },
        new Category { Id = "3", Name = "Vegetables" },
        new Category { Id = "4", Name = "Bakery" },
        new Category { Id = "5", Name = "Meat" },
    ];

    public List<Category> GetAllCategories() => _categories;

    public Category? GetCategoryById(string id) => _categories.FirstOrDefault(c => c.Id == id);
}
